README.md

```diff
@@ -70,6 +70,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 *Tools that provide metrics and quality measurements.*
 
 * [Checkstyle](http://checkstyle.sourceforge.net/) - Static analysis of coding conventions and standards.
+* [ErrorProne](https://github.com/google/error-prone) - Catch common Java mistakes as compile-time errors from Google.
```

Comment:
To me the description sounds correct. It is what they describe it them selfs. They catch mistakes which go beyond compiler warnings and halt the build if they are severe. See example [SelfAssignment](http://errorprone.info/bugpattern/SelfAssignment)